### PR TITLE
Node: Add binary variant to `PUBSUB NUMSUB` and `PUBSUB SHARDNUMSUB`

### PR DESCRIPTION
**Pull request by yipin-chen**

### CODE REVIEW

**Comment by yipin-chen** [comment]
@Yury-Fridlyand: for these 2 commands, I am not sure how to verify binary in the test. Any suggestion?

File: node/src/BaseClient.ts [hunk 1]
```diff
@@ -6874,12 +6877,11 @@ export class BaseClient {
      * ```
      */
     public async pubsubNumSub(
-        channels: string[],
-        options?: DecoderOption,
+        options?: { channels?: GlideString[] } & DecoderOption,
```
Comment (Yury-Fridlyand) [inline node/src/BaseClient.ts:6880]: I'd keep channels mandatory arg, so we can remove them from options

File: node/src/BaseClient.ts [hunk 2]
```diff
@@ -6874,12 +6877,11 @@ export class BaseClient {
...
     ): Promise<{ channel: GlideString; numSub: number }[]> {
         return this.createWritePromise<GlideRecord<number>>(
-            createPubSubNumSub(channels),
-            options,
+            createPubSubNumSub(options?.channels),
+            { decoder: options?.decoder },
```
Comment (Yury-Fridlyand) [inline node/src/BaseClient.ts:6884]: Anyway you can simplify that
```suggestion
            options,
```

File: node/src/GlideClusterClient.ts [hunk 3]
```diff
@@ -1257,28 +1258,36 @@ export class GlideClusterClient extends BaseClient {
...
-     * console.log(result2); // Output: {}
+     * const result2 = await client.pubsubShardnumsub([]);
+     * console.log(result2); // Output: []
      * ```
      */
     public async pubsubShardNumSub(
-        channels?: string[],
-    ): Promise<Record<string, number>> {
-        return this.createWritePromise(createPubSubShardNumSub(channels));
+        options?: { channels?: GlideString[] } & DecoderOption,
```
Comment (Yury-Fridlyand) [inline node/src/GlideClusterClient.ts:1281]: same here

File: node/src/GlideClusterClient.ts [hunk 4]
```diff
@@ -1257,28 +1258,36 @@ export class GlideClusterClient extends BaseClient {
     /**
      * Returns the number of subscribers (exclusive of clients subscribed to patterns) for the specified shard channels.
      *
-     * Note that it is valid to call this command without channels. In this case, it will just return an empty map.
-     * The command is routed to all nodes, and aggregates the response to a single map of the channels and their number of subscriptions.
-     *
      * @see {@link https://valkey.io/commands/pubsub-shardnumsub/|valkey.io} for details.
+     * @remarks When in cluster mode, the command is routed to all nodes, and aggregates the response into a single list.
```
Comment (Yury-Fridlyand) [inline node/src/GlideClusterClient.ts:1262]: This command exists on cluster client only
```suggestion
     * @remarks The command is routed to all nodes, and aggregates the response into a single list.
```

File: node/src/GlideClusterClient.ts [hunk 5]
```diff
@@ -1257,28 +1258,36 @@ export class GlideClusterClient extends BaseClient {
...
      */
     public async pubsubShardNumSub(
-        channels?: string[],
-    ): Promise<Record<string, number>> {
-        return this.createWritePromise(createPubSubShardNumSub(channels));
+        options?: { channels?: GlideString[] } & DecoderOption,
+    ): Promise<{ channel: GlideString; numSub: number }[]> {
+        return this.createWritePromise<GlideRecord<number>>(
+            createPubSubShardNumSub(options?.channels),
+            { decoder: options?.decoder },
```
Comment (Yury-Fridlyand) [inline node/src/GlideClusterClient.ts:1285]:
```suggestion
            options,
```

File: node/src/GlideClusterClient.ts [hunk 6]
```diff
@@ -1257,28 +1258,36 @@ export class GlideClusterClient extends BaseClient {
...
-     *
      * @see {@link https://valkey.io/commands/pubsub-shardnumsub/|valkey.io} for details.
+     * @remarks When in cluster mode, the command is routed to all nodes, and aggregates the response into a single list.
      *
-     * @param channels - The list of shard channels to query for the number of subscribers.
-     *                   If not provided, returns an empty map.
-     * @returns A map where keys are the shard channel names and values are the number of subscribers.
+     * @param options - (Optional) Additional parameters:
+     * - (Optional) `channels`: the list of shard channels to query for the number of subscribers.
+     *     If not provided, returns an empty map.
```
Comment (Yury-Fridlyand) [inline node/src/GlideClusterClient.ts:1266]: You can remove this if arg is mandatory.
```suggestion
```

File: node/src/Transaction.ts [hunk 7]
```diff
@@ -4176,16 +4176,14 @@ export class ClusterTransaction extends BaseTransaction<ClusterTransaction> {
     /**
      * Returns the number of subscribers (exclusive of clients subscribed to patterns) for the specified shard channels.
      *
-     * Note that it is valid to call this command without channels. In this case, it will just return an empty map.
-     * The command is routed to all nodes, and aggregates the response to a single map of the channels and their number of subscriptions.
-     *
      * @see {@link https://valkey.io/commands/pubsub-shardnumsub|valkey.io} for more details.
+     * @remarks When in cluster mode, the command is routed to all nodes, and aggregates the response into a single list.
```
Comment (Yury-Fridlyand) [inline node/src/Transaction.ts:4180]: apply same changes in transaction

**Review by Yury-Fridlyand — COMMENTED** [review]
There is no way to test binary options unless we implement GlideStrings into pubsub config.

**Review by Yury-Fridlyand — APPROVED** [review]

File: node/tests/TestUtilities.ts [hunk 8]
```diff
@@ -689,8 +689,6 @@ export async function transactionTest(
     responseData.push(["pubsubChannels()", []]);
     baseTransaction.pubsubNumPat();
     responseData.push(["pubsubNumPat()", 0]);
-    baseTransaction.pubsubNumSub();
```
Comment (Yury-Fridlyand) [inline node/tests/TestUtilities.ts:692]: You can keep `baseTransaction.pubsubNumSub([]);`

Comment (yipin-chen) [reply node/tests/TestUtilities.ts:692]: I removed `Optional` channel in Transaction. It will fail if I revert this changes.

**Pull request merged**